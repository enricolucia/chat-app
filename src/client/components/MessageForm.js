import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const FormContainer = styled.div`
  display: flex;
  justify-content: space-between;
`

const StyledInput = styled.input`
  flex: 1 1 85%;
  padding: 0 10px;
  line-height: 40px;
  border: 0;
`

const StyledButton = styled.button`
  flex: 1;
  background: #265778;
  color: #FFFFFF;
  font-weight: bold;
  border: 0;
`

class MessageForm extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      value: ''
    }
    this.onSubmit = this.onSubmit.bind(this)
    this.onChange = this.onChange.bind(this)
  }

  onChange () {
    this.setState({
      value: this.inputElement.value
    })
  }

  onSubmit (e) {
    if ((!e.keyCode || e.keyCode === 13) && this.inputElement.value) {
      this.props.onInteraction(this.state.value.trim())
      this.setState({
        value: ''
      })
    }
  }

  render () {
    const { value } = this.state
    return (
      <FormContainer>
        <StyledInput type='text'
          onChange={this.onChange}
          innerRef={ref => { this.inputElement = ref }}
          onKeyUp={this.onSubmit}
          value={value}
        />
        <StyledButton onClick={this.onSubmit}>
          {String.fromCharCode(8682)}
        </StyledButton>
      </FormContainer>
    )
  }
}

MessageForm.propTypes = {
  onInteraction: PropTypes.func
}

export default MessageForm
