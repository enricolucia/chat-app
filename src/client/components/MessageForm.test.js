/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'

import MessageForm from './MessageForm'

const defaultProps = {
  onInteraction: jest.fn(x => x)
}

describe('MessageForm', () => {
  it('renders message from', () => {
    const tree = renderer.create(
      <MessageForm {...defaultProps} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
