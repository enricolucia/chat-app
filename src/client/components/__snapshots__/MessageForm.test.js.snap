// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageForm renders message from 1`] = `
<div
  className="MessageForm__FormContainer-jGJyiO bzhYoL"
>
  <input
    className="MessageForm__StyledInput-ldUKTA hfglTZ"
    onChange={[Function]}
    onKeyUp={[Function]}
    type="text"
    value=""
  />
  <button
    className="MessageForm__StyledButton-gFwgwZ dVRzac"
    onClick={[Function]}
  >
    â‡ª
  </button>
</div>
`;
